nmt_chainer/beam_search.py

```diff
@@ -105,6 +105,19 @@ def compute_next_lists(new_state_ensemble, new_scores, beam_width, eos_idx, curr
 #             if len(next_states_list) >= beam_width:
 #                 break
 
+    # Prune items that have a score worse than beam_size.
+    if (consider_beam_size_when_pruning and next_score_list):
+        best_next_score = max(next_score_list)
+        bad_score_indices = [idx for idx, elem in enumerate(next_score_list) if (best_next_score - elem > beam_size)]
+        wc = len(next_words_list)
+
+        for i in sorted(bad_score_indices, reverse=True):
```

Comment:
Actually, the indices in bad_score_indices should already be sorted in ascending order. So if you want them in descending order, you can just use "for i in bad_score_indices[::-1]:"